terraform/main.tf

```diff
@@ -8,36 +8,30 @@ module "networking" {
   resource_group_name = var.resource_group_name
   location            = var.location
   address_space       = var.address_space
-  subnets             = var.subnets
+  virtual_machines    = var.virtual_machines
 }
 
 module "security" {
   source              = "./modules/security"
-  nsg_name            = var.nsg_name
   resource_group_name = var.resource_group_name
   location            = var.location
-}
 
-module "rcb_vm" {
-  source               = "./modules/vm"
-  resource_group_name  = var.resource_group_name
-  location             = var.location
-  server_name          = var.rcb_vm_name
-  admin_username       = var.admin_username
-  admin_ssh_public_key = var.admin_ssh_public_key
-  disk_size            = var.disk_size
-  subnet_id            = module.networking.subnet_ids[0]
+  security_groups = [for vm in var.virtual_machines : vm.networking.network_interface.security_group]
 }
 
-module "exm_vm" {
-  source               = "./modules/vm"
-  resource_group_name  = var.resource_group_name
-  location             = var.location
-  server_name          = var.exm_vm_name
+module "vm" {
+  count = length(var.virtual_machines)
```

Comment:
Cred ca aici ar fi mai bine sa facem cu for_each = var.virtual_machines (o sa se creeze exact cate vms sunt in lista/map, iar daca lista/mapul e gol atunci nu face nimic)